Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,284 @@
+      "Collateral Asset - Collateral-3":
+        "0x0000000000000000000000000000000000001289",
+      "Collateral Asset - Collateral-4":
+        "0x0000000000000000000000000000000000001290",
+    };
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(1);
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: `Change of asset values due to governance proposal`,
+        description: `The asset ${mockAssetTokenAddress} has been modified by a governance proposal`,
+        alertId: "GOV-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: txEvent.network.toString(),
+        metadata: {
+          ...metadata1,
+          ...metadata,
+        },
+      }),
+    ]);
+
+    mockProvider.clear();
+    expect(findings.length).toStrictEqual(1);
+  });
+
+  it("should return 0 findings for event other than proposal change", async () => {
+    mockProvider.setNetwork(1);
+    txEvent.addEventLog(
+      "event mockEvent(address value1, uint8 value2)",
+      CONFIGURATOR_PROXY,
+      [createAddress("0x9467"), 2],
+    );
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(0);
+  });
+
+  it("should return 0 findings for multiple non-governance events", async () => {
+    mockProvider.setNetwork(1);
+    const nonGovEvents = [
+      "event Transfer(address indexed from, address indexed to, uint256 value)",
+      "event Approval(address indexed owner, address indexed spender, uint256 value)",
+      "event Deposit(address indexed user, address reciepient, uint256 amount)",
+    ];
+
+    nonGovEvents.forEach((eventSignature) => {
+      txEvent.addEventLog(eventSignature, createAddress("0xABCDEF"), [
+        createAddress("0x12345"),
+        createAddress("0x67890"),
+        1000,
+      ]);
+    });
+
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(0);
+  });
+
+  it("should a finding for a governance event out of multiple non-governance events", async () => {
+    mockProvider.setNetwork(1);
+    const nonGovEvents = [
+      "event Transfer(address indexed from, address indexed to, uint256 value)",
+      "event Approval(address indexed owner, address indexed spender, uint256 value)",
+      "event Deposit(address indexed user, address reciepient, uint256 amount)",
+    ];
+    nonGovEvents.forEach((eventSignature) => {
+      txEvent.addEventLog(eventSignature, createAddress("0xABCDEF"), [
+        createAddress("0x12345"),
+        createAddress("0x67890"),
+        1000,
+      ]);
+    });
+
+    txEvent.addEventLog(
+      "event SetGovernor(address indexed cometProxy, address indexed oldGovernor, address indexed newGovernor)",
+      CONFIGURATOR_PROXY,
+      [...mockArgs],
+    );
+    const findings = await handleTransaction(txEvent);
+    expect(findings.length).toStrictEqual(1);
```

Comment:
Team standard practice to check for the actual alert that is being created and not just checking that there is one. With this check, it won't tell you that it is the one you expect, but simply that one was returned. Please update.